File: mailbox-android/src/main/java/org/briarproject/mailbox/android/dontkillme/HuaweiProtectedAppsView.java
package org.briarproject.mailbox.android.dontkillme;

import android.content.Context;
import android.util.AttributeSet;

import org.briarproject.android.dontkillmelib.PowerUtils;
import org.briarproject.mailbox.R;

import androidx.annotation.Nullable;
import androidx.annotation.StringRes;
import androidx.annotation.UiThread;

import static org.briarproject.android.dontkillmelib.PowerUtils.getHuaweiProtectedAppsIntent;

@UiThread
public class HuaweiProtectedAppsView extends PowerView {

	public HuaweiProtectedAppsView(Context context) {
		this(context, null);
	}

	public HuaweiProtectedAppsView(Context context,
			@Nullable AttributeSet attrs) {
		this(context, attrs, 0);
	}

	public HuaweiProtectedAppsView(Context context,
			@Nullable AttributeSet attrs,
			int defStyleAttr) {
		super(context, attrs, defStyleAttr);
		setText(R.string.dnkm_huawei_protected_text);
		setButtonText(R.string.dnkm_huawei_protected_button);
	}

	@Override
	public boolean needsToBeShown() {
		return PowerUtils.huaweiProtectedAppsNeedsToBeShown(getContext());
	}

	@Override
	@StringRes
	protected int getHelpText() {
		return R.string.dnkm_huawei_protected_help;
	}

	@Override
	protected void onButtonClick() {
		getContext().startActivity(getHuaweiProtectedAppsIntent());
		setChecked(true);
	}

}

File: mailbox-android/src/main/java/org/briarproject/mailbox/android/dontkillme/HuaweiProtectedAppsView.kt
/*
 *     Briar Mailbox
 *     Copyright (C) 2021-2022  The Briar Project
 *
 *     This program is free software: you can redistribute it and/or modify
 *     it under the terms of the GNU Affero General Public License as
 *     published by the Free Software Foundation, either version 3 of the
 *     License, or (at your option) any later version.
 *
 *     This program is distributed in the hope that it will be useful,
 *     but WITHOUT ANY WARRANTY; without even the implied warranty of
 *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *     GNU Affero General Public License for more details.
 *
 *     You should have received a copy of the GNU Affero General Public License
 *     along with this program.  If not, see <https://www.gnu.org/licenses/>.
 *
 */

package org.briarproject.mailbox.android.dontkillme

import android.content.Context
import android.util.AttributeSet
import androidx.annotation.UiThread
import org.briarproject.android.dontkillmelib.HuaweiUtils.huaweiProtectedAppsIntent
import org.briarproject.android.dontkillmelib.HuaweiUtils.protectedAppsNeedsToBeShown
import org.briarproject.mailbox.R

@UiThread
internal class HuaweiProtectedAppsView @JvmOverloads constructor(
    context: Context,
    attrs: AttributeSet? = null,
    defStyleAttr: Int = 0,
) : PowerView(context, attrs, defStyleAttr) {

    init {
        setText(R.string.dnkm_huawei_protected_text)
        setIcon(R.drawable.ic_security_white)
        setButtonText(R.string.dnkm_huawei_protected_button)
    }

    override fun needsToBeShown(): Boolean {
        return protectedAppsNeedsToBeShown(context)
    }

    override val helpText: Int = R.string.dnkm_huawei_protected_help

    override fun onButtonClick() {
        context.startActivity(huaweiProtectedAppsIntent)
        setChecked(true)
    }
}
